### ISSUE TITLE
Support for converting cases

### ISSUE THREAD

**Issue by ppakseev** [dus4nstojanovic/create-files-from-template-base#1]
Hey @dus4nstojanovic ,

What do you think about adding support for converting file name to different case?
I'm thinking about adding some additional attributes to the file templates and paths to make it work.

**Example with FileName = "demo element"**: 
```ts
// path: component#(kebab-case).ts --> demo-element.ts

export const DEFAULT_FileName#(PascalCase,UPPERCASE)_PROP = ... --> DEFAULT_DEMO_ELEMENT_PROP = ...

const FileName#(PascalCase) = ...  --> DemoElement = ...
```

I assume this library might be used for automatic conversion, so only a few changes for pattern-matching are required.
https://www.npmjs.com/package/js-convert-case

Just asking your opinion on integrating this feature into the library.
If you don't mind I can implement this feature by myself. Or you can lead the development yourself)

**Comment by dus4nstojanovic** [comment]
@ppakseev I like the idea.

The package needs to support something like this. Feel free to create the pull request if you already know how to do it. Otherwise, I will investigate and implement it. It shouldn't be hard. 

I can later extend the Visual Studio code extension and CLI to support the same. 👍

**Comment by dus4nstojanovic** [comment]
Hey @ppakseev!

I have some time to work on the proposed feature today. The new version will be available today/tomorrow for both the CLI and VS Code extension.

**Comment by paufau** [comment]
Hi @dus4nstojanovic,
Awesome! Appreciate your effort!

I was just starting to look at where to start adding the feature, and I noticed that you already added it :)

**Comment by paufau** [comment]
I see you decided to use UPPER_SNAKE_CASE for enum CaseOption. What do you think about making the enum values represent the case itself. 

I think this will make the case definition more visually clear. Perhaps this is done by the first parameter that needs to be passed before specifying the case, since I haven't yet had time to understand the implementation details)

**Comment by DusanStojanovic-gowago** [comment]
I am not sure I understand. Can you give an example?

**Comment by paufau** [comment]
Now, to convert the string to the desired case, I have to write it like this, right?
`const #(FileName,PASCAL_CASE) = ...  --> const DemoElement = ...`
or
`const #(FileName, KEBAB_CASE) = ... --> const demo-component = ...`

I'm talking about making the case variable name reflect its behavior.
So it would be:
`const #(FileName,PascalCase) = ...  --> const DemoElement = ...`
or
`const #(FileName, kebab-case) = ... --> const demo-component = ...`

**Comment by DusanStojanovic-gowago** [comment]
I understand. Hmmm... I get what you think and it makes sense. But for some people it would be hard to remember the keys in that case.

I am also thinking about making the keys (enum values) case insensitive.

For example: PascalCase, pascalcase, or pascalCase, PASCALCASE.

What do you think?

**Comment by paufau** [comment]
This is just a wonderful idea! 
It is important not to forget to remove all special characters to make "snake_case" and "SNAKECASE" work similarly

**Comment by dus4nstojanovic** [comment]
I published new **CLI (1.7.3)** and **VSCode extension (1.7.2)** versions. I also updated the `Readme.md` with the new feature!

Here are the release notes:
- [VSCode extension](https://github.com/dus4nstojanovic/create-files-from-template-vscode/releases/tag/v1.7.2)
- [CLI](https://github.com/dus4nstojanovic/create-files-from-template-cli/releases/tag/1.7.3)

Thank you one more time for suggesting this feature!

**Comment by paufau** [comment]
Amazing! Thanks for implementing it so quickly)
